### PR TITLE
implements CREATE INDEX statement

### PR DESCRIPTION
**Pull request by SUMUKHA-PK**
This PR completes the implementation of CREATE INDEX stmt (https://sqlite.org/lang_createindex.html) and is a part of #66 

This PR also fixes delimiting in scanning keywords - lets keywords to end on all non-alphanumeric values instead of only whitespaces.

Includes a parsing logic for separating `CREATE xxx` statements (with function skeletons for other create xxx functions).

### CODE REVIEW

File: internal/parser/parser_test.go [hunk 1]
```diff
@@ -474,6 +906,14 @@ func TestSingleStatementParse(t *testing.T) {
...
 						return false
 					}
+					// fmt.Println(t1.Col())
+					// fmt.Println(t2.Col())
+					// fmt.Println(t1.Length())
+					// fmt.Println(t2.Length())
+					// fmt.Println(t1.Offset())
+					// fmt.Println(t2.Offset())
+					// fmt.Println(t1.Type())
+					// fmt.Println(t2.Type())
```
Comment (TimSatke) [inline internal/parser/parser_test.go]:
```suggestion
```

File: internal/parser/simple_parser_rules.go [hunk 2]
```diff
@@ -50,6 +50,10 @@ func (p *simpleParser) parseSQLStatement(r reporter) (stmt *ast.SQLStmt) {
 		stmt.BeginStmt = p.parseBeginStmt(r)
 	case token.KeywordCommit:
 		stmt.CommitStmt = p.parseCommitStmt(r)
+	case token.KeywordCreate:
+		// for now we can directly add create index as the parsing function
+		// later there must be an intermediate layer to take care of different CREATE functions
+		stmt.CreateIndexStmt = p.parseCreateIndexStmt(r)
```
Comment (TimSatke) [inline internal/parser/simple_parser_rules.go]: Do this now.

File: internal/parser/simple_parser_rules.go [hunk 3]
```diff
@@ -929,3 +933,157 @@ func (p *simpleParser) parseRollbackStmt(r reporter) (stmt *ast.RollbackStmt) {
...
+	}
+	if next.Type() == token.KeywordWhere {
+		stmt.Where = next
+		p.consumeToken()
+		stmt.Expr = p.parseExpression(r)
+	}
+	return
+}
+
+func (p *simpleParser) parseIndexedColumns(r reporter) (stmt *ast.IndexedColumn) {
```
Comment (TimSatke) [inline internal/parser/simple_parser_rules.go]: That's not what the spec says. https://www.sqlite.org/lang_createindex.html

Comment (TimSatke) [inline internal/parser/simple_parser_rules.go]: Does it really parse many?

File: internal/parser/parser_test.go [hunk 4]
```diff
@@ -452,6 +452,438 @@ func TestSingleStatementParse(t *testing.T) {
...
+			"CREATE INDEX myIndex ON myTable (exprLiteral)",
+			&ast.SQLStmt{
+				CreateIndexStmt: &ast.CreateIndexStmt{
+					Create:    token.New(1, 1, 0, 6, token.KeywordCreate, "CREATE"),
+					Index:     token.New(1, 8, 7, 5, token.KeywordIndex, "INDEX"),
+					IndexName: token.New(1, 14, 13, 7, token.Literal, "myIndex"),
+					On:        token.New(1, 22, 21, 2, token.KeywordOn, "ON"),
+					TableName: token.New(1, 25, 24, 7, token.Literal, "myTable"),
+					LeftParen: token.New(1, 33, 32, 1, token.Delimiter, "("),
+					IndexedColumns: []*ast.IndexedColumn{
```
Comment (TimSatke) [inline internal/parser/parser_test.go]: Add tests with multiple `IndexedColumn`s

File: internal/parser/simple_parser_rules.go (hunk 3 above)
Comment (SUMUKHA-PK) [reply internal/parser/simple_parser_rules.go]: What exactly are you talking about?

Comment (SUMUKHA-PK) [reply internal/parser/simple_parser_rules.go]: You mean separated by commas? Probably not, still a WIP

File: internal/parser/parser_test.go (hunk 4 above)
Comment (SUMUKHA-PK) [reply internal/parser/parser_test.go]: Will do, still a WIP

File: internal/parser/simple_parser_rules.go (hunk 2 above)
Comment (SUMUKHA-PK) [reply internal/parser/simple_parser_rules.go]: Sure. How would you do this? I will need to peek the next element?

File: internal/parser/parser_test.go (hunk 1 above)
Comment (SUMUKHA-PK) [reply internal/parser/parser_test.go]: Ugh my bad

File: internal/parser/simple_parser_rules.go (hunk 2 above)
Comment (TimSatke) [reply internal/parser/simple_parser_rules.go]: ParserStmt -> parseCreateStmt -> create index, table etc

File: internal/parser/simple_parser_rules.go (hunk 3 above)
Comment (TimSatke) [reply internal/parser/simple_parser_rules.go]: IndexedColumn is more than just an expression

Comment (SUMUKHA-PK) [reply internal/parser/simple_parser_rules.go]: Yes, I havent added the rest.

Comment (SUMUKHA-PK) [reply internal/parser/simple_parser_rules.go]: This parses one single Indexed column and is appended later above. Ill change the name to `parseIndexedColumn`

File: internal/parser/parser_test.go (hunk 4 above)
Comment (SUMUKHA-PK) [reply internal/parser/parser_test.go]: Done

File: internal/parser/simple_parser_rules.go (hunk 2 above)
Comment (SUMUKHA-PK) [reply internal/parser/simple_parser_rules.go]: Done

File: internal/parser/simple_parser_rules.go (hunk 3 above)
Comment (SUMUKHA-PK) [reply internal/parser/simple_parser_rules.go]: Done

File: internal/parser/scanner/ruleset/ruleset_default.go [hunk 5]
```diff
@@ -71,7 +71,7 @@ func defaultKeywordsRule(s RuneScanner) (token.Type, bool) {
 	var buf bytes.Buffer
 	for {
 		next, ok := s.Lookahead()
-		if !ok || defaultWhitespaceDetector.Matches(next) {
+		if !ok || defaultWhitespaceDetector.Matches(next) || !defaultLiteral.Matches(next) {
```
Comment (TimSatke) [inline internal/parser/scanner/ruleset/ruleset_default.go]: Is the whitespace check really neccessary any more?

File: internal/parser/simple_parser_rules.go [hunk 6]
```diff
@@ -1082,8 +1142,65 @@ func (p *simpleParser) parseCreateIndexStmt(r reporter) (stmt *ast.CreateIndexSt
...
+func (p *simpleParser) parseCreateTriggerStmt(createToken, tempToken, temporaryToken token.Token, r reporter) (stmt *ast.CreateTriggerStmt) {
+	return
+}
+
+func (p *simpleParser) parseCreateViewStmt(createToken, tempToken, temporaryToken token.Token, r reporter) (stmt *ast.CreateViewStmt) {
+	return
+}
+
+func (p *simpleParser) parseCreateVirtualTableStmt(createToken token.Token, r reporter) (stmt *ast.CreateVirtualTableStmt) {
 	return
```
Comment (TimSatke) [inline internal/parser/simple_parser_rules.go]: Still WIP?

Comment (SUMUKHA-PK) [reply internal/parser/simple_parser_rules.go]: No, these are for the next issues
You can see that they were added to create the parsing logic for check statements.

File: internal/parser/scanner/ruleset/ruleset_default.go (hunk 5 above)
Comment (SUMUKHA-PK) [reply internal/parser/scanner/ruleset/ruleset_default.go]: Ah yes, can be removed

**Pull request merged**